src/_nebari/stages/kubernetes_services/template/modules/kubernetes/services/conda-store/config/conda_store_config.py

```diff
@@ -83,11 +88,301 @@ def conda_store_config(path="/var/lib/conda-store/config.json"):
 c.GenericOAuthAuthentication.user_data_key = "preferred_username"
 c.GenericOAuthAuthentication.tls_verify = False
 
+CONDA_STORE_ROLE_PERMISSIONS_ORDER = ["viewer", "developer", "admin"]
+
+
+@dataclasses.dataclass
+class CondaStoreNamespaceRole:
+    namespace: str
+    role: str
+
+
+@dataclasses.dataclass
+class KeycloakCondaStoreRoleScopes:
+    scopes: str
+    log: logging.Logger
+
+    def _validate_role(self, role):
+        valid = role in CONDA_STORE_ROLE_PERMISSIONS_ORDER
+        self.log.info(f"role: {role} is valid={valid}")
+        return valid
+
+    def parse_role_and_namespace(
+        self, text
+    ) -> typing.Optional[CondaStoreNamespaceRole]:
+        # The regex pattern
+        pattern = r"^(\w+)!namespace=([^!]+)$"
+
+        # Perform the regex search
+        match = re.search(pattern, text)
+
+        # Extract the permission and namespace if there is a match
+        if match and self._validate_role(match.group(1)):
+            return CondaStoreNamespaceRole(
+                namespace=match.group(2), role=match.group(1)
+            )
+        else:
+            return None
+
+    def parse_scope(self) -> typing.List[CondaStoreNamespaceRole]:
+        """Parsed scopes from keycloak role's attribute and returns a list of role/namespace
+        if scopes' syntax is valid otherwise return []
+
+        Example:
+            Given scopes as "viewer!namespace=scipy,admin!namespace=pycon", the function will
+            return [{"role": "viewer", "namespace": "scipy"}, {"role": "admin", "namespace": "pycon"}]
+        """
+        if not self.scopes:
+            self.log.info(f"No scope found: {self.scopes}, skipping role")
+            return []
+        scope_list = self.scopes.split(",")
+        parsed_scopes = []
+        self.log.info(f"Scopes to parse: {scope_list}")
+        for scope_text in scope_list:
+            parsed_scope = self.parse_role_and_namespace(scope_text)
+            parsed_scopes.append(parsed_scope)
+            if not parsed_scope:
+                self.log.info(f"Unable to parse: {scope_text}, skipping keycloak role")
+                return []
+        return parsed_scopes
+
 
 class KeyCloakAuthentication(GenericOAuthAuthentication):
+    conda_store_api_url = f"https://{config['external-url']}/conda-store/api/v1"
+    access_token_url = config["openid-config"]["token_url"]
+    realm_api_url = config["realm_api_url"]
+    service_account_token = config["service-tokens-mapping"][
+        "conda-store-service-account"
+    ]
+
+    def _disable_ssl_verify_ctx(self):
+        """This is useful for deployments without certs, otherwise conda-store login would
+        not work.
+        """
+        import ssl
+
+        ctx = ssl.create_default_context()
+        ctx.check_hostname = False
+        ctx.verify_mode = ssl.CERT_NONE
+        return ctx
```

Comment:
It seems like we shouldn't disable ssl verification for all Nebari deployments, but only those without certs since you are vulnerable to MITM attacks by disabling it.  Maybe it can be justified if we were using the k8s svc dns instead of the <nebari domain>/auth endpoint.  What do you think?